### DISCUSSION TITLE
[Adonis 4.1] Database schema with zerofill auto_increments?

### DISCUSSION THREAD

**Discussion by aprakasa** [adonisjs/core#1467] [Help] [answered]
```js
'use strict'

/** @type {import('@adonisjs/lucid/src/Schema')} */
const Schema = use('Schema')

class CardSchema extends Schema {
  up () {
    this.create('cards', (table) => {
      table.increments() // How to add zerofill in here or
      table.int('cardID', 4).increments() // here
      table.timestamps()
    })
  }

  down () {
    this.drop('cards')
  }
}

module.exports = CardSchema
```

I want the output into something like this:

| ID | cardID |
| --- | --- |
|  1 | 0001 |
|  2 | 0002 |
|  3 | 0003 |
|  4 | 0004 |
|  5 | 0005 |

**Comment by thetutlage** [comment, marked as answer]
I don't know about zerofill. Maybe point me to the SQL documentation for it?

**Reply by aprakasa** [reply]
Originally was google it from here https://stackoverflow.com/a/5256505 and https://mariadb.com/kb/en/int/

**Reply by thetutlage** [reply]
You have to run a raw query to alter the column and add `ZEROFILL`.


```js
// not tested
this.raw('ALTER TABLE cards CHANGE reference id INT(2) UNSIGNED ZEROFILL NOT NULL')
```

As a side note, I am not sure this is something you should be using coz of the following reasons

- The leading zeros in Javascript are removed from the number data type. For JS `00001 === 1`.
- I think MYSQL understands this is an issue and that's why there docs mention depreciating `zerofill` with int data type

<img width="944" alt="CleanShot 2020-08-16 at 12 36 01@2x" src="https://user-images.githubusercontent.com/1706381/90328824-28877380-dfbd-11ea-864c-0e50f9278673.png">

**Reply by aprakasa** [reply]
Ahh didn't read that part. Thank you for the enlightenment
